check50/__main__.py

```diff
@@ -278,87 +313,97 @@ def main():
 
-
+    # dev implies offline and verbose "info" if not overwritten
     if args.dev:
         args.offline = True
-        args.verbose = True
+        if not args.verbose:
+            args.verbose = "info"
 
+    # offline implies local
     if args.offline:
+        args.no_install_dependencies = True
+        args.no_download_checks = True
         args.local = True
 
+    # Setting any verbosity level implies logs from checks
     if args.verbose:
-        # Show lib50 commands being run in verbose mode
-        logging.basicConfig(level=os.environ.get("CHECK50_LOGLEVEL", "INFO"))
-        lib50.ProgressBar.DISABLED = True
         args.log = True
 
+    # Setup logging for lib50 depending on verbosity level
+    setup_logging(args.verbose)
+
+    # Warning in case of running remotely with no_download_checks or no_install_dependencies set
+    if (args.no_download_checks or args.no_install_dependencies) and not args.local:
+        if args.no_download_checks and args.no_install_dependencies:
+            msg = _("--no-downloads-checks and --no-install-dependencies have")
+        elif args.no_download_checks:
+            msg = _("--no-downloads-checks has")
+        else:
+            msg = _("--no-install-dependencies has")
+        termcolor.cprint(_("Warning: {} no effect without --local").format(msg), "yellow", attrs=["bold"])
+
     # Filter out any duplicates from args.output
     seen_output = set()
     args.output = [output for output in args.output if not (output in seen_output or seen_output.add(output))]
 
     # Set excepthook
-    excepthook.verbose = args.verbose
+    excepthook.verbose = bool(args.verbose)
     excepthook.outputs = args.output
     excepthook.output_file = args.output_file
 
+    # If remote, push files to GitHub and await results
     if not args.local:
         commit_hash = lib50.push("check50", SLUG, internal.CONFIG_LOADER, data={"check50": True})[1]
         with lib50.ProgressBar("Waiting for results") if "ansi" in args.output else nullcontext():
             tag_hash, results = await_results(commit_hash, SLUG)
+    # Otherwise run checks locally
     else:
-        with lib50.ProgressBar("Checking") if not args.verbose and "ansi" in args.output else nullcontext():
+        with lib50.ProgressBar("Checking") if "ansi" in args.output else nullcontext():
             # If developing, assume slug is a path to check_dir
             if args.dev:
                 internal.check_dir = Path(SLUG).expanduser().resolve()
                 if not internal.check_dir.is_dir():
                     raise internal.Error(_("{} is not a directory").format(internal.check_dir))
+            # Otherwise have lib50 create a local copy of slug
             else:
-                # Otherwise have lib50 create a local copy of slug
                 try:
-                    internal.check_dir = lib50.local(SLUG, offline=args.offline)
+                    internal.check_dir = lib50.local(SLUG, offline=args.no_download_checks)
```

Comment:
Any reason we're not passing `offline=args.offline` instead of specifically `args.no_download_checks` here?

Reply:
because `no_download_checks` does not imply offline

Reply:
Oh you're right!